pandas/tests/frame/test_arithmetic.py

```diff
@@ -642,3 +642,18 @@ def test_arith_non_pandas_object(self):
         val3 = np.random.rand(*df.shape)
         added = pd.DataFrame(df.values + val3, index=df.index, columns=df.columns)
         tm.assert_frame_equal(df.add(val3), added)
+
+    def test_operations_with_interval_categories_index(self, all_arithmetic_operators):
+        # GH#27415
+        op = all_arithmetic_operators
+        ind = pd.CategoricalIndex(pd.interval_range(start=0.0, end=2.0))
+
+        df = pd.DataFrame([[1, 2]], columns=ind)
+        num = 100
+        try:
+            getattr(df, op)(num)
```

Comment:
these need to compare the result, if there is an exception it will *already* be raised so we don't need to specifically fail.

your code should look like

```
result = getattr(df, op((num)
expected = 
tm.assert_frame_equal(result, expected)
```

Reply:
Thanks a lot for the suggestion! I though I only need to check for the raise of TypeError. Now I have added the expected result and assertion.